bin/ContributeRecipe

```diff
@@ -218,7 +218,10 @@ then
    git stash pop
    git branch -D "submit-$program-$version"
 else
+   git config --global hub.protocol https
    hub --version &> /dev/null || Die "You need to install 'hub' to auto-submit pull-requests"
-   hub pull-request -b "$compileUpstreamBranch" -h "submit-$program-$version" -m "$program $version"
+   hub fork --remote-name origin-fork
+   git push --set-upstream  origin-fork "submit-$program-$version"
+   hub pull-request -b "origin/$compileUpstreamBranch" -h "origin-fork/submit-$program-$version" -m "$program $version"
 fi
```

Comment:
Finally we are left with two choices: 
1) Checkout back to main and reapply latest user's commit as 
2) Stay on user-modified branch and update places where recipe repo is fetched to automatically rebase this branch on top of newly pulled updates.

Reply:
Of these two, I'd go with option 2. 

> Maybe a good idea would be to have a separate folder for user-generated recipes, so that those conflicts are resolved on fielsystem level. That would only require a check if both upstream and user-defined versions exists there should be a prompt to ask user which one to use during Compile

We used to have something like that up until 016: before we adopted git for /D/C/Recipes, there was also a /D/C/LocalRecipes. The back and forth between them was kind of annoying (but then, we have commit access to the repo so we may be biased). But generally I don't expect conflicts to be frequent, since each package and version live in their own folder already. A conflict most likely would mean a submitted recipe which was superseded by another submission that was taken into the official repo instead.

In any case, even if we had a second git repo with the user's fork at /D/C/LocalRecipes, if a conflict arises between the Recipes master and the user's fork, that will eventually hit their fork when they try to update to run ContributeRecipes next time. So I'm not sure if having two trees really gains us much.